### ISSUE TITLE
is_hash, is_array, and is_any are depricated

### ISSUE THREAD

**Issue by james-powis** [voxpupuli/puppet-gitlab#106]
reference: https://github.com/puppetlabs/puppetlabs-stdlib
and example of replacement: https://github.com/puppetlabs/puppetlabs-ntp/blob/master/manifests/init.pp#L17-L28

**Comment by tobru** [comment]
Thanks! We'll update the module over the next weeks...

**Comment by iainhallam** [comment]
Same for validate_bool, apparently (at least, I get warnings) in Puppet agent > 4.

**Comment by pgassmann** [comment]
For reference and search, the deprecation messages:
method is deprecated, please use the stdlib validate_legacy function

```
Warning: This method is deprecated, please use the stdlib validate_legacy function, with Stdlib::Compat::Bool. There is further documentation for validate_legacy function in the README. at ["/x/modules/gitlab/manifests/init.pp", 339]:["/x/site/role/manifests/gitlab.pp", 2]
   (at /x/modules/stdlib/lib/puppet/functions/deprecation.rb:25:in `deprecation')
Warning: This method is deprecated, please use the stdlib validate_legacy function, with Stdlib::Compat::String. There is further documentation for validate_legacy function in the README. at ["/x/modules/gitlab/manifests/init.pp", 342]:["/x/site/role/manifests/gitlab.pp", 2]
   (at /x/modules/stdlib/lib/puppet/functions/deprecation.rb:25:in `deprecation')
Warning: This method is deprecated, please use the stdlib validate_legacy function, with Pattern[]. There is further documentation for validate_legacy function in the README. at ["/x/modules/gitlab/manifests/init.pp", 344]:["/x/site/role/manifests/gitlab.pp", 2]
   (at /x/modules/stdlib/lib/puppet/functions/deprecation.rb:25:in `deprecation')
Warning: This method is deprecated, please use the stdlib validate_legacy function, with Stdlib::Compat::Absolute_Path. There is further documentation for validate_legacy function in the README. at ["/x/modules/gitlab/manifests/init.pp", 351]:["/x/site/role/manifests/gitlab.pp", 2]
   (at /x/modules/stdlib/lib/puppet/functions/deprecation.rb:25:in `deprecation')
Warning: This method is deprecated, please use the stdlib validate_legacy function, with Stdlib::Compat::Hash. There is further documentation for validate_legacy function in the README. at ["/x/modules/gitlab/manifests/init.pp", 364]:["/x/site/role/manifests/gitlab.pp", 2]
   (at /x/modules/stdlib/lib/puppet/functions/deprecation.rb:25:in `deprecation')
```

**Comment by iainhallam** [comment]
Was this fixed, then?

**Comment by LongLiveCHIEF** [comment]
yep! I was going through all issues last night to make sure we cleaned any up, and I couldn't remember the exact commit/merge where this was fully resolved or I would have linked it. I'm fairly sure though that the first release containing this fix was in `v1.16.0`.
